cupy/_core/core.pyx

```diff
@@ -91,6 +92,18 @@ cdef tuple _HANDLED_TYPES
 
 cdef object _null_context = contextlib.nullcontext()
 
+cdef bint _is_sam_enabled = (int(os.environ.get('CUPY_ENABLE_SAM', '0')) != 0)
```

Comment:
I think the renaming from HMM (what we had) to SAM is problematic, as SAM is specific to only `malloc` while HMM can be also used to refer to managed memory (which I didn't have time to cover). However I'll let go for now.

Reply:
I think is better to decide a name and stick to it now than change it later and having "deprecated" options

Reply:
How about `CUPY_ENABLE_UMP`? UMP stands for "Unified Memory Programming", which puts emphasis on not only the change of underlying memory resource (ex: CuPy users can already use managed memory today), but also the change of the programming model (no distinction in memory space anymore, only in execution space).

https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#unified-memory-programming

Reply:
I like this

Reply:
changed to UMP in [846682f](https://github.com/cupy/cupy/pull/8442/commits/846682f1d23deaf109b3edbf7d49fb5f520d735e)